server/src/main/java/io/spine/server/entity/Transaction.java

```diff
@@ -445,6 +439,61 @@ public void setDeleted(boolean deleted) {
     }
 
     /**
+     * Retrieves the {@link VersionAdvancementStrategy} of current {@code Transaction}.
+     *
+     * <p>The value should be constant along all the instances of a certain (runtime) type.
+     *
+     * <p>The default value is {@link VersionAdvancementStrategy#INJECT INJECT}.
+     */
+    protected VersionAdvancementStrategy versioningStrategy() {
+        return INJECT;
+    }
+
+    /**
+     * The strategy of advancing the version of a {@code Transaction} upon the given {@link Phase}.
+     */
+    protected enum VersionAdvancementStrategy {
+
+        /**
+         * Retrieves the version from the {@linkplain EventContext event}, which caused
+         * the transaction phase.
+         *
+         * @see EventContext#getVersion()
+         */
+        INJECT {
+            @Override
+            Version advanceUpon(Phase<?, ?, ?, ?> phase) {
+                final Version fromEvent = phase.getContext().getVersion();
+                return fromEvent;
+            }
+        },
+
+        /**
+         * Retrieves the current version of the {@code Transaction} incremented by one.
```

Comment:
The description must be better. This is an enum member. It does not retrieve anything. Same applies to the previous enumeration member.